x-pack/test/apm_api_integration/tests/services/throughput.ts

```diff
@@ -47,93 +47,94 @@ export default function ApiTest({ getService }: FtrProviderContext) {
     });
   });
 
-  let throughputResponse: ThroughputReturn;
+  async function callThroughApis() {
+    const path = {
+      serviceName: 'opbeans-go',
+    };
+    const commonQuery = {
+      start: metadata.start,
+      end: metadata.end,
+      transactionType: 'request',
+      environment: 'ENVIRONMENT_ALL',
+    };
+    const [throughputMetricsResponse, throughputTransactionsResponse] = await Promise.all([
+      apmApiSupertest({
+        endpoint: 'GET /api/apm/services/{serviceName}/throughput',
+        params: {
+          path,
+          query: {
+            ...commonQuery,
+            kuery: 'processor.event : "metric"',
+          },
+        },
+      }),
+      apmApiSupertest({
+        endpoint: 'GET /api/apm/services/{serviceName}/throughput',
+        params: {
+          path,
+          query: {
+            ...commonQuery,
+            kuery: 'processor.event : "transaction"',
+          },
+        },
+      }),
+    ]);
+    return {
+      throughputMetrics: throughputMetricsResponse.body,
+      throughputTransactions: throughputTransactionsResponse.body,
+    };
+  }
+
   registry.when(
     'Throughput when data is loaded',
     { config: 'basic', archives: [archiveName] },
     () => {
-      describe('when querying without kql filter', () => {
+      let throughputMetrics: ThroughputReturn;
+      let throughputTransactions: ThroughputReturn;
+
+      describe('compare metrics and transactions values', () => {
         before(async () => {
-          const response = await apmApiSupertest({
-            endpoint: 'GET /api/apm/services/{serviceName}/throughput',
-            params: {
-              path: {
-                serviceName: 'opbeans-java',
-              },
-              query: {
-                start: metadata.start,
-                end: metadata.end,
-                transactionType: 'request',
-                environment: 'ENVIRONMENT_ALL',
-                kuery: '',
-              },
-            },
-          });
-          throughputResponse = response.body;
+          const throughputApiResponse = await callThroughApis();
+          throughputMetrics = throughputApiResponse.throughputMetrics;
+          throughputTransactions = throughputApiResponse.throughputTransactions;
         });
 
-        it('returns some data', () => {
-          expect(throughputResponse.currentPeriod.length).to.be.greaterThan(0);
-          expect(throughputResponse.previousPeriod.length).not.to.be.greaterThan(0);
-
-          const nonNullDataPoints = throughputResponse.currentPeriod.filter(({ y }) =>
-            isFiniteNumber(y)
+        it('returns some transactions data', () => {
+          expect(throughputTransactions.currentPeriod.length).to.be.greaterThan(0);
+          const nonNullTransactionsDataPoints = throughputTransactions.currentPeriod.filter(
+            ({ y }) => isFiniteNumber(y)
           );
```

Comment:
Looks like you just need a boolean answer, not a number. In that case `some` is a better fit

```suggestion
          const hasData = throughputTransactions.currentPeriod.some(
            ({ y }) => isFiniteNumber(y)
          );
```